### ISSUE TITLE
Unable to import existing project (terraform code downloaded from console setup process)

### ISSUE THREAD

**Issue by olvesh** [terraform-google-modules/terraform-google-project-factory#722]
### TL;DR

- "Set up Google Cloud for your organization" via IAM in console created Terraform project. 
- This project failed due to billing account quotas 
- When continuing (after increasing quotas) Terraform fails with `Error 409: Requested entity already exists`  (it is my project)
- On import google-project-factory fails as described below

### Expected behavior

I expected to be able to import my project which is not in state file from previous erroneous attempt and continue the setup.

### Observed behavior

Note that `myproject-log-prod` is not the real name. I do own and control the real project and am aware that the resources need to be globally unique.

```
❯ terraform import module.myproject-log-prod.module.project-factory.google_project.main myproject-log-prod
module.myproject-log-prod.module.project-factory.google_project.main: Importing from ID "myproject-log-prod"...
module.myproject-log-prod.module.project-factory.google_project.main: Import prepared!
  Prepared google_project for import
module.myproject-log-prod.module.project-factory.google_project.main: Refreshing state... [id=projects/myproject-log-prod]
╷
│ Error: Invalid index
│
│   on .terraform/modules/myproject-log-prod/modules/core_project_factory/main.tf line 39, in locals:
│   39:     google_service_account.default_service_account[0].email,
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/myproject-log-prod/modules/core_project_factory/outputs.tf line 37, in output "service_account_id":
│   37:   value       = var.create_project_sa ? google_service_account.default_service_account[0].account_id : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/myproject-log-prod/modules/core_project_factory/outputs.tf line 42, in output "service_account_display_name":
│   42:   value       = var.create_project_sa ? google_service_account.default_service_account[0].display_name : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/myproject-log-prod/modules/core_project_factory/outputs.tf line 47, in output "service_account_email":
│   47:   value       = var.create_project_sa ? google_service_account.default_service_account[0].email : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/myproject-log-prod/modules/core_project_factory/outputs.tf line 52, in output "service_account_name":
│   52:   value       = var.create_project_sa ? google_service_account.default_service_account[0].name : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/myproject-log-prod/modules/core_project_factory/outputs.tf line 57, in output "service_account_unique_id":
│   57:   value       = var.create_project_sa ? google_service_account.default_service_account[0].unique_id : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-nonprod/modules/core_project_factory/main.tf line 39, in locals:
│   39:     google_service_account.default_service_account[0].email,
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-nonprod/modules/core_project_factory/outputs.tf line 37, in output "service_account_id":
│   37:   value       = var.create_project_sa ? google_service_account.default_service_account[0].account_id : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-nonprod/modules/core_project_factory/outputs.tf line 42, in output "service_account_display_name":
│   42:   value       = var.create_project_sa ? google_service_account.default_service_account[0].display_name : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-nonprod/modules/core_project_factory/outputs.tf line 47, in output "service_account_email":
│   47:   value       = var.create_project_sa ? google_service_account.default_service_account[0].email : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-nonprod/modules/core_project_factory/outputs.tf line 52, in output "service_account_name":
│   52:   value       = var.create_project_sa ? google_service_account.default_service_account[0].name : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-nonprod/modules/core_project_factory/outputs.tf line 57, in output "service_account_unique_id":
│   57:   value       = var.create_project_sa ? google_service_account.default_service_account[0].unique_id : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-prod/modules/core_project_factory/main.tf line 39, in locals:
│   39:     google_service_account.default_service_account[0].email,
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-prod/modules/core_project_factory/outputs.tf line 37, in output "service_account_id":
│   37:   value       = var.create_project_sa ? google_service_account.default_service_account[0].account_id : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-prod/modules/core_project_factory/outputs.tf line 42, in output "service_account_display_name":
│   42:   value       = var.create_project_sa ? google_service_account.default_service_account[0].display_name : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-prod/modules/core_project_factory/outputs.tf line 47, in output "service_account_email":
│   47:   value       = var.create_project_sa ? google_service_account.default_service_account[0].email : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-prod/modules/core_project_factory/outputs.tf line 52, in output "service_account_name":
│   52:   value       = var.create_project_sa ? google_service_account.default_service_account[0].name : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵

╷
│ Error: Invalid index
│
│   on .terraform/modules/qt-mon-prod/modules/core_project_factory/outputs.tf line 57, in output "service_account_unique_id":
│   57:   value       = var.create_project_sa ? google_service_account.default_service_account[0].unique_id : ""
│     ├────────────────
│     │ google_service_account.default_service_account is empty tuple
│
│ The given key does not identify an element in this collection value: the collection has no elements.
╵
```

### Terraform Configuration

```hcl
I am using the downloaded zip-file from the IAM setup created by google cloud for new organisations.
```

### Terraform Version

```sh
❯ terraform version
Terraform v1.1.9
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.20.0
+ provider registry.terraform.io/hashicorp/google-beta v4.20.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.3
+ provider registry.terraform.io/hashicorp/time v0.7.2
```

### Additional information

By setting `create_project_sa = false` in the projects I was able to continue, and setting it to true after resolved the issue..

Just submitting this, and will close soon if there is not something to be done, but ok to document if someone else stumbles over the same issue.
